### PR TITLE
Fix docs to focus more on eStargz

### PR DESCRIPTION
**Pull request by ktock**
#184

This includes the following changes
- fixing the docs to focus on eStargz
- updating the benchmark result to the recent one
- fixing README to make each sentence start from the new line

### CODE REVIEW

File: docs/stargz-estargz.md [hunk 1]
```diff
@@ -1,81 +1,223 @@
...
 
-## Stargz archive format
+## Overview
 
-When pulling an image lazily, the runtime doesn’t download the entire contents but fetches necessary data chunks *on-demand*.
-For archiving this, runtimes need to *selectively* fetch and extract files in the image layers.
-However current OCI Image Spec uses tar and tar.gz archive formats for archiving layers, which don't suit to this use-case because of the following reasons,
+When lazily pulling an image from the registry, necessary chunks of its layers are fetched *on-demand* during running the container, instead of downloading the entire contents of that image at once.
+For achieving this, runtimes need to *selectively* fetch and extract files contents in the layer.
+However current OCI/Docker Image Spec uses tar (optionally with compression) for archiving layers, which doesn't suit to this use-case because of the following reasons,
```
Comment (AkihiroSuda) [inline docs/stargz-estargz.md:17]: However -> However,

File: docs/stargz-estargz.md [hunk 2]
```diff
@@ -1,81 +1,223 @@
...
+  - `char`: character device
+  - `block`: block device
+  - `fifo`: fifo
+  - `chunk`: a chunk of regular file data
+  As described in the above section, a regular file can be divided into several chunks.
+  Corresponding to the first chunk of that file, TOCEntry typed `reg` MUST be contained.
+  Corresponding to the chunks after 2nd, TOCEntries typed `chunk` MUST be contained.
+  `chunk`-typed TOCEntry must set offset, chunkOffset and chunkSize properties.
+
+- **`size`** *int64*
```
Comment (AkihiroSuda) [inline docs/stargz-estargz.md:99]: Isn't this uint64?

File: docs/stargz-estargz.md [hunk 3]
```diff
@@ -1,81 +1,223 @@
...
+
+- **`linkName`** *string*
+
+  This OPTIONAL property contains the link target of `symlink` and `hardlink`.
+
+- **`mode`** *int64*
+
+  This OPTIONAL property contains the permission and mode bits.
+
+- **`uid`** *int*
```
Comment (AkihiroSuda) [inline docs/stargz-estargz.md:117]: uint

File: docs/stargz-estargz.md [hunk 4]
```diff
@@ -1,81 +1,223 @@
...
+This MUST be an empty gzip member ([RFC1952](https://tools.ietf.org/html/rfc1952)) whose [Extra field](https://tools.ietf.org/html/rfc1952#section-2.3.1.1) contains the offset of TOC in the archive.
+The footer MUST be the following 47 bytes (1 byte = 8 bits in gzip).
+
+```
+10 bytes  gzip header
+2  bytes  XLEN (length of Extra field) = 22 (16 hex digits + len("STARGZ"))
+22 bytes  Extra field (fmt.Sprintf("%016xSTARGZ", offsetOfTOC))
+5  bytes  flate header
+8  bytes  gzip footer
+(End of the eStargz blob)
```
Comment (AkihiroSuda) [inline docs/stargz-estargz.md:185]: Is this section about plain stargz or eStargz?

File: docs/stargz-estargz.md (hunk 1 above)
Comment (ktock) [reply docs/stargz-estargz.md]: Fixed this.

File: docs/stargz-estargz.md (hunk 2 above)
Comment (ktock) [reply docs/stargz-estargz.md]: Fixed this. The implementation will be following-up PRs.

File: docs/stargz-estargz.md (hunk 3 above)
Comment (ktock) [reply docs/stargz-estargz.md]: Fixed this. The implementation will be following-up PRs.

File: docs/stargz-estargz.md (hunk 4 above)
Comment (ktock) [reply docs/stargz-estargz.md]: Fixed the footer section to make it compliant to [Extra field in RFC 1952](https://tools.ietf.org/html/rfc1952#section-2.3.1.1) (thus eStargz's footer).
The implementation will be the following-up PR.

File: docs/stargz-estargz.md [hunk 5]
```diff
@@ -1,81 +1,227 @@
...
+- 2  bytes  XLEN (length of Extra field) = 26 (4 bytes header + 16 hex digits + len("STARGZ"))
+- 2  bytes  Extra: SI1 = 'S', SI2 = 'G'
+- 2  bytes  Extra: LEN = 22 (16 hex digits + len("STARGZ"))
+- 22 bytes  Extra: subfield = fmt.Sprintf("%016xSTARGZ", offsetOfTOC)
+- 5  bytes  flate header
+- 8  bytes  gzip footer
+(End of eStargz)
+```
+
+*NOTE that though eStarz is compatible to tar.gz, it's not compatible to stargz by Google CRFS.*
```
Comment (AkihiroSuda) [inline docs/stargz-estargz.md:189]: Needs full explanation. The footer differs (for RFC compliance) but TOC is compatible, right?

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: docs/stargz-estargz.md (hunk 5 above)
Comment (ktock) [reply docs/stargz-estargz.md]: Yes. Added `Notes on compatibility with stargz` subsection for explaining this.

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**